### PR TITLE
Add the Doxygen configuration

### PR DESCRIPTION
**Pull request by nomis52**
@peternewman @joshua-moyerman

### CODE REVIEW

File: src/transport.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+
+#ifndef SRC_TRANSPORT_H_
+#define SRC_TRANSPORT_H_
+
+#include <stdint.h>
+#include "constants.h"
+
+/**
+ * @brief A IOVector, with a pointer to memory and a length attribute.
```
Comment (peternewman) [inline src/transport.h]: SPaG An

File: tests/Doxyfile [hunk 2]
```diff
@@ -0,0 +1,1854 @@
+# Doxyfile 1.8.3.1
```
Comment (peternewman) [inline tests/Doxyfile]: Why's this all in tests, or is it a limitation of the build and not polluting the uC codespace?

**Comment by peternewman** [comment]
LGTM

File: src/transport.h (hunk 1 above)
Comment (nomis52) [inline src/transport.h]: Done

File: tests/Doxyfile (hunk 2 above)
Comment (nomis52) [inline tests/Doxyfile]: Yes. The top level folder is managed by Microchip's MPLAB.

**Pull request merged**